### DISCUSSION TITLE
Does eglot ever interact with dired for move refactorings or any other types of refactorings (eglot-rename etc...)?

### DISCUSSION THREAD

**Discussion by jgarte** [joaotavora/eglot#975] [Q&A]
Does eglot ever interact with dired for move refactorings or any other types of refactorings (eglot-rename etc...)?

See https://github.com/pappasam/jedi-language-server/issues/209

**Comment by joaotavora** [comment]
No, not directly.  There is not a single mention of `dired` in `eglot.el`.  Eglot itself only edits files.  Language servers that have access to the file system (this is the majority that I know of) may create/delete files when they so choose.  In current Eglot, I think they can inform Eglot's user that they have done so.

**Reply by jgarte** [reply]
@joaotavora Thanks for the info. Yes, I saw no direct mention of any `dired-*` functions in the code base. I didn't check if any functions indirectly called `dired-*` functions though...

Have you used any of the [move refactorings](https://www.jetbrains.com/help/pycharm/move-refactorings.html) with any language server to move modules with eglot before?

I haven't been able to do it yet. 

[jedi-language-server](https://github.com/pappasam/jedi-language-server) supports it as pappasam confirmed and maybe [python-lsp-server](https://github.com/python-lsp/python-lsp-server) with the [rope plugin](https://github.com/python-rope/pylsp-rope).

But, I think pappasam tests mostly on neovim with jedi-language-server...

**Reply by joaotavora** [reply]
> Have you used any of the [move refactorings](https://www.jetbrains.com/help/pycharm/move-refactorings.html) with any language server to move modules with eglot before?

No.

> [jedi-language-server](https://github.com/pappasam/jedi-language-server) supports it as pappasam confirmed and maybe [python-lsp-server](https://github.com/python-lsp/python-lsp-server) with the [rope plugin](https://github.com/python-rope/pylsp-rope).

Eglot only speaks LSP, it doesn't speak Jedi, papasam or rope. So if you can point us in the direction of a particular LSP operation(s) in the LSP specification that are used in the mentioned file-manupulating refactorings, we can look at implementing it.
